### PR TITLE
Sonarqube

### PR DESCRIPTION
**Pull request by jbakosi**
WIP: Removed ubuntu-sonarqube and added alpine dockerfile that is ready for sonarqube. I managed to successfully do a sonarqube analysis inside this alpine image by hand and upload to https://sonarqube.com/dashboard?id=quinoa. Still need to
- [x] configure/finetune the sonarqube analysis, e.g., inclusions, exclusions, etc.
- [x] hook up to travis
- [x] rebase this branch to exclude ubuntu-image attempt before merge
- [x] do not forget to switch branches at docker hub for the automated image build once merged
- [x] set `sonar.projectVersion` to git commit sha1, see also https://docs.sonarqube.org/display/SONAR/Analysis+Parameters

### CODE REVIEW

**Review by junghans — APPROVED** [review]
Looks good to me!

Should we try to hook it up as an auto-build in docker hub?

File: alpine [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# Configure user
+RUN adduser -S quinoa quinoa
+USER quinoa
+WORKDIR /home/quinoa
+
+# Clone quinoa
+RUN git clone https://github.com/quinoacomputing/quinoa.git
+# Build TPLs
+RUN cd quinoa && mkdir -p tpl/build && cd tpl/build && cmake -DCMAKE_CXX_COMPILER=mpicxx -DCMAKE_C_COMPILER=mpicc -DCMAKE_Fortran_COMPILER=mpif90 -DCMAKE_INSTALL_PREFIX=/home/quinoa/tpl .. && make -sj$((`cat /proc/cpuinfo | grep MHz | wc -l`))
```
Comment (junghans) [inline alpine]: `grep -c processor /proc/cpuinfo` might be easier!

**Comment by jbakosi** [comment]
It is already hooked up from this branch. I'm struggling to keep it under the max 2 hours limit. This may also improve with quinoa-ng.

File: alpine (hunk 1 above)
Comment (jbakosi) [reply alpine]: Sure!

**Pull request merged**